promql/quantile.go

```diff
@@ -83,6 +83,7 @@ func bucketQuantile(q float64, buckets buckets) float64 {
 		return math.NaN()
 	}
 
+	buckets = coalesceBuckets(buckets)
```

Comment:
This is done after the `len(buckets) > 2` check, but `coalesceBuckets()` may remove buckets, causing the number of buckets to be <2 again. It seems due to the extra `+Inf` check, this can only happen if the input bucket list is an edge case like this:

`["+inf", "+Inf"]`

Not sure if the following code then actually has problems with this, but adding a test case for this would be good.

Reply:
Good point, fixed.